tools/ci-scripts/linux/image-cleanup/collector/live_images.py

```diff
@@ -0,0 +1,153 @@
+ImageTag = NewType('ImageTag', str) # pylint: disable=invalid-name
+
+def _blobs_from_roots(
+        repo_path: str,
+        roots: _RevListRoots) -> AbstractSet[_BlobId]:
+    """Return the blob IDs of the revisions of .travis.yml for all the commits
+specified by the given `roots`.
+
+    :param repo_path: Path to the repository to inspect.
+
+    :param roots: A collection of argument lists to pass to ``git
+    rev-list`` to limit the matched commits.
+
+    :return The blob IDs of the .travis.yml revisions.
+    """
+
+    commit_ids = set()
+    for root_args in roots:
+        git_rev_list_cmd_line = ['git', '-C', repo_path, 'rev-list']
+        git_rev_list_cmd_line.extend(root_args)
+
+        logger.debug('Invoking %s', git_rev_list_cmd_line)
+
+        rev_list_output = subprocess.check_output(git_rev_list_cmd_line, stderr=subprocess.PIPE)
+        commit_ids.update(str(rev_list_output, 'utf-8').splitlines())
+        logger.debug('Commit IDs currently: %s', commit_ids)
+
+    blob_ids = set()
+    for commit in commit_ids:
+        git_ls_tree_cmd_line = ['git',
+                                '-C', repo_path,
+                                'ls-tree', commit,
+                                '--', ':/.travis.yml']
+        logger.debug('Invoking %s', git_ls_tree_cmd_line)
+        ls_tree_output = subprocess.check_output(git_ls_tree_cmd_line, stderr=subprocess.PIPE)
+
+        for line in str(ls_tree_output, 'utf-8').splitlines():
+            logger.debug('Parsing ls-tree output: "%s"', line)
+
+            # Expecing output like:
+            #
+            # 100644 blob cbe47c031fb164dca034aac640de08cf35170a68	.travis.yml
+            parts = line.split()
+
+            # The length of parts may not be exactly 4 if the name
+            # contains a space. (Though, this shouldn't happen in this
+            # particular case.) Instead of parsing exactly, we check
+            # that we have at least the first three parts and call it
+            # good enough.
+            if len(parts) < 3:
+                raise ValueError('Cannot parse ls-tree output: "{}"'.format(line))
+
+            obj_type = parts[1]
+            blob_id = _BlobId(parts[2])
+            if obj_type == 'blob':
+                blob_ids.add(blob_id)
+
+    logger.debug('Blobs: %s', blob_ids)
+    return frozenset(blob_ids)
+
+def _images_from_blobs(
+        repo_path: str,
+        commits: Iterable[_BlobId]) -> AbstractSet[ImageName]:
+    """Get the full names (including tags) of the images used for the
+CI build in the given blobs.
+
+    :param repo_path: Path to the repository to inspect.
+
+    :param commits: The Git blob IDs of files in which to look for
+    image names. It's assumed that these are blob IDs of .travis.yml or
+    similar files.
+
+    :return The full image names used by the specificed blobs.
+
+    Images are found by looking for a line like
+
+    CI_BUILD_CONTAINER=...
```

Comment:
More work: Let's rename this in `.travis.yml` and here in this PR.

Reply:
✔️